### ISSUE TITLE
Add folder to sync workflow is counter intuitive

### ISSUE THREAD

**Issue by jnweiger** [owncloud/client#3756]
I have removed the root folder sync, and can now use the button 'Add Folder to Synchronize'
When clicking this button, I
- first chosse a local folder and can optionally give a descriptive name
- then second browse the server's folder hierarchy for the folder to connect with.

I suggest to reverse that workflow and first choose a folder on the server.
This has multiple advantages:
- The exact spelling of the folder name can be suggested. No more guessing trial/error needed.
- The location of the folder can be suggested to mimic the hierarchical structure as seen on the server. Currently user has to navigate starting from a seemingly random location. 
- When making multiple folder sync connections, the last local filesystem location should be remembered, and could be offered later instead of keeping an unrelated (random) location as a starting point.

**Comment by jnweiger** [comment]
The last button on the last dialog window is 'Add Folder'. It should be called 'Connect Folder'.

**Comment by guruz** [comment]
@jancborchardt

**Comment by jancborchardt** [comment]
Yup, agree with @jnweiger’s suggestion. However this will of course mess with the local-first approach @dragotin favors.
